.github/workflows/main.yml

```diff
@@ -45,51 +45,51 @@ jobs:
         #- target: mipsel-unknown-linux-gnu
         #  os: ubuntu-latest
         #  rust: nightly
-        - target: powerpc-unknown-linux-gnu
-          os: ubuntu-latest
-          rust: nightly
-        - target: powerpc64-unknown-linux-gnu
-          os: ubuntu-latest
-          rust: nightly
-        - target: powerpc64le-unknown-linux-gnu
-          os: ubuntu-latest
-          rust: nightly
+        #- target: powerpc-unknown-linux-gnu
+        #  os: ubuntu-latest
+        #  rust: nightly
+        #- target: powerpc64-unknown-linux-gnu
+        #  os: ubuntu-latest
+        #  rust: nightly
+        #- target: powerpc64le-unknown-linux-gnu
+        #  os: ubuntu-latest
+        #  rust: nightly
         - target: sbf-solana-solana
           os: ubuntu-latest
           rust: nightly
-        - target: thumbv6m-none-eabi
-          os: ubuntu-latest
-          rust: nightly
-        - target: thumbv7em-none-eabi
-          os: ubuntu-latest
-          rust: nightly
-        - target: thumbv7em-none-eabihf
-          os: ubuntu-latest
-          rust: nightly
-        - target: thumbv7m-none-eabi
-          os: ubuntu-latest
-          rust: nightly
-        - target: wasm32-unknown-unknown
-          os: ubuntu-latest
-          rust: nightly
-        - target: x86_64-unknown-linux-gnu
-          os: ubuntu-latest
-          rust: nightly
-        - target: x86_64-apple-darwin
-          os: macos-latest
-          rust: nightly
-        - target: i686-pc-windows-msvc
-          os: windows-latest
-          rust: nightly
-        - target: x86_64-pc-windows-msvc
-          os: windows-latest
-          rust: nightly
-        - target: i686-pc-windows-gnu
-          os: windows-latest
-          rust: nightly-i686-gnu
-        - target: x86_64-pc-windows-gnu
-          os: windows-latest
-          rust: nightly-x86_64-gnu
+        #- target: thumbv6m-none-eabi
+        #  os: ubuntu-latest
+        #  rust: nightly
+        #- target: thumbv7em-none-eabi
+        #  os: ubuntu-latest
+        #  rust: nightly
+        #- target: thumbv7em-none-eabihf
+        #  os: ubuntu-latest
+        #  rust: nightly
+        #- target: thumbv7m-none-eabi
+        #  os: ubuntu-latest
+        #  rust: nightly
+        #- target: wasm32-unknown-unknown
+        #  os: ubuntu-latest
+        #  rust: nightly
+        #- target: x86_64-unknown-linux-gnu
+        #  os: ubuntu-latest
+        #  rust: nightly
+        #- target: x86_64-apple-darwin
+        #  os: macos-latest
+        #  rust: nightly
+        #- target: i686-pc-windows-msvc
+        #  os: windows-latest
+        #  rust: nightly
+        #- target: x86_64-pc-windows-msvc
+        #  os: windows-latest
+        #  rust: nightly
+        #- target: i686-pc-windows-gnu
+        #  os: windows-latest
+        #  rust: nightly-i686-gnu
+        #- target: x86_64-pc-windows-gnu
+        #  os: windows-latest
+        #  rust: nightly-x86_64-gnu
     steps:
```

Comment:
why is it a problem to check we don't break compiler-builtins for other platforms?

Reply:
I removed them because they fail:

```
error: unexpected `cfg` condition value: `solana`
   --> testcrate/tests/misc.rs:138:5
    |
138 |     target_os = "solana"
    |     ^^^^^^^^^^^^^^^^^^^^
    |
    = note: expected values for `target_os` are: `aix`, `android`, `cuda`, `dragonfly`, `emscripten`, `espidf`, `freebsd`, `fuchsia`, `haiku`, `hermit`, `horizon`, `hurd`, `illumos`, `ios`, `l4re`, `linux`, `macos`, `netbsd`, `none`, `nto`, `nuttx`, `openbsd`, `psp`, `psx`, `redox`, `rtems`, `solaris`, `solid_asp3`, `teeos`, `trusty`, `tvos`, `uefi`, `unknown`, `visionos`, and `vita` and 6 more
```

Reply:
maybe we should add solana to known target_os or use a different condition to disable this test for solana, for example `target_vendor = "solana"` might work? Is the expected value checked by rust compiler or is the check logic somewhere in the test harness? in any case it's better to fix this rather than disable CI for all supported targets, it seems.

Reply:
I found a solution here: https://github.com/anza-xyz/compiler-builtins/pull/27

I'll push a couple more commits to this PR before requesting your review again.

Reply:
Ready now.